### ISSUE TITLE
[Bug] Session participants are not updated after the call ends

### ISSUE THREAD

**Issue by vihan85** [GetStream/stream-video-js#1561]
**Which package/packages do you use?**

- [x] `@stream-io/video-react-sdk`
- [ ] `@stream-io/video-react-native-sdk`
- [ ] `@stream-io/video-client`

**Describe the bug**
Session participants are not updated after the call ends

**To Reproduce**

1. User 1  create call with two participants
2. user 1 leave call
3. user 2 end call
4. user 1 call  useCallSession()  and see  useCallSession().participants result

**Screenshots**
[If applicable, add screenshots to help explain your problem.
](https://drive.google.com/drive/folders/167QQMg3l8bb18NhBo0wH6wbMxGlnSwFi?usp=sharing)
**Version**

**Expected behavior**
After end call useCallSession().participants = []

- SDK version: @stream-io/video-react-sdk: 1.7.9

**Comment by oliverlaz** [comment]
hi @vihan85, thanks for reporting this issue. This behavior is intentional and standardized across all of our SDKs.
We are evaluating whether to introduce a flag that enables state updates even after the call transitions to LEFT state. I'll keep you posted on this.

Can you please share what's your use case? Maybe we can suggest another solution.

**Comment by vihan85** [comment]
hi @oliverlaz,

I'm developing a functionality when during a group call, users can join whenever they want by clicking the "join call" button, and the button will disappear if the call ends.

- "join call" button just appear while  useCallSession().participants.length > 0

**Comment by Qodestackr** [comment]
"@stream-io/node-sdk": "^0.2.5",
    "@stream-io/video-react-sdk": "^1.1.3",
    
    For me, a blank screen is presented...

**Comment by vihan85** [comment]
Hi @oliverlaz,

Can you help me?

**Comment by Qodestackr** [comment]
[client]: Failed to connect Error: {"code":"","StatusCode":"","message":"initial WS connection could not be established","isWSFailure":true}

I keep getting this error, esp in production

**Comment by Qodestackr** [comment]
Hello @oliverlaz

**Comment by oliverlaz** [comment]
hi @vihan85,

We can't change this behavior as it might be a breaking change for some customers who rely on the existing behavior.
We might consider adding an option on the `call` instance that can control this behavior in the future.

Currently, I can suggest you one workaround:
```ts
const onLeaveCallButtonPress = async () => {
  await call.leave(); // I assume this is your current logic
  
  // Add this:
  // This will re-establish the connection and will enable state updates until you do `call.leave()` again.
  // Keep in mind this may create a memory leak, as this call instance won't be automatically disposed.
  await call.get();
}
```

**Comment by oliverlaz** [comment]
@Qodestackr, can you please open another issue?

**Comment by oliverlaz** [comment]
Hi @vihan85, I'm closing this issue due to inactivity. I hope the last solution provided works for you.
If it doesn't, please reopen this issue.

**Comment by vihan85** [comment]
hi @oliverlaz 

I tested your solution but it doesn't work correctly with some members who haven't joined the call yet.

**Comment by oliverlaz** [comment]
hi @vihan85, can you explain in a bit more detail what doesn't work?

**Comment by vihan85** [comment]
Hi @oliverlaz 

I used. 
```js const onLeaveCallButtonPress = async () => {
  await call.leave(); // I assume this is your current logic
  
  // Add this:
  // This will re-establish the connection and will enable state updates until you do `call.leave()` again.
  // Keep in mind this may create a memory leak, as this call instance won't be automatically disposed.
  await call.get();
}

For group call participants: after user A joined then leave (call onLeaveCallButtonPress ) and the call ends, useCallSession().participants.length will equal 0. However, for group members who have not yet joined the call, useCallSession().participants.length will remain 1.

**Comment by vihan85** [comment]
Hi @oliverlaz 

Are you still here?

**Comment by oliverlaz** [comment]
hi @vihan85,

I'm afraid I can't reproduce the issues you reported. Please check the attached video and the companion code sandbox:
- https://codesandbox.io/p/sandbox/stream-video-session-participants-6g3mcg

https://github.com/user-attachments/assets/f8640800-5c41-4df1-8583-82c70324115c

It would be great if you could prepare a code sample that I can use to reproduce the issues you have.

**Comment by vihan85** [comment]
Hi @oliverlaz 

I ended call but  useCallSession().participants.length still > 0

https://github.com/user-attachments/assets/52e64a7d-4a0f-44cc-bbb3-068208427a83

**Comment by vihan85** [comment]
> hi [@vihan85](https://github.com/vihan85), `call.endCall()` triggers `call.leave()` under the hood for every call participant. For remotely ended calls, you can re-subscribe for state updates like this:
> 
> const off = call.on('call.ended', async (e) => {
>   if (call.state.callingState === CallingState.LEFT) {
>     await call.get();
>   }
> });
> 
> off(); // to unsubscribe

Hi @oliverlaz 

I used your solution but it incorrect with my context. After I call  call.get(), a call appears in useCalls() hook with the callingState 'ringing'. This is not correct because I want to update the number of call participants to 0 after the call ends and will no longer watch this call until I use the getOrCreate function again.

**Comment by vihan85** [comment]
hi @oliverlaz  

Are you still here?

[this is demo 
](https://codesandbox.io/p/sandbox/stream-video-session-participants-forked-vgvxx9?file=%2Fsrc%2FApp.tsx)

**Comment by oliverlaz** [comment]
hi @vihan85, apologies for the late response. I think I've found the bug, and I'll let you know once a fix is released.

**Comment by oliverlaz** [comment]
hey @vihan85, this issue should be now fixed as part of `@stream-io/video-react-sdk@1.12.2`.
Can you please check if everything works as expected now?
